### ISSUE TITLE
question about learning rate scheduling

### ISSUE THREAD

**Issue by wingsyuan** [marian-nmt/marian-dev#567]
Hi,

When I use the default parameters  to train an en-zh model with a trainning dataset containing 10m pairs:
--learn-rate 0.0003 --lr-warmup 16000 --lr-decay-inv-sqrt 16000 --lr-report \

the training process showed the bleu score on raw validation scores grow really slowly after 100000 updates:

[2019-12-07 13:39:22] [valid] Ep. 7 : Up. 129000 : perplexity : 43.0322 : new best
[2019-12-07 13:40:00] [valid] Ep. 7 : Up. 129000 : translation : 14.39 : stalled 1 times (last best: 14.46)
[2019-12-07 13:44:46] Ep. 7 : Up. 129500 : Sen. 2,569,702 : Cost 2.45853496 : Time 363.89s : 18381.54 words/s : L.r. 1.0545e-04
[2019-12-07 13:49:30] Ep. 7 : Up. 130000 : Sen. 2,779,122 : Cost 2.46268439 : Time 283.88s : 23664.70 words/s : L.r. 1.0525e-04
[2019-12-07 13:54:21] Ep. 7 : Up. 130500 : Sen. 3,016,484 : Cost 2.45618677 : Time 290.72s : 24080.71 words/s : L.r. 1.0505e-04
[2019-12-07 13:54:21] Saving model weights and runtime parameters to model_gradual_2/ens/model.npz.orig.npz
[2019-12-07 13:54:28] Saving model weights and runtime parameters to model_gradual_2/ens/model.npz
[2019-12-07 13:54:35] Saving Adam parameters to model_gradual_2/ens/model.npz.optimizer.npz
[2019-12-07 13:54:50] Saving model weights and runtime parameters to model_gradual_2/ens/model.npz.best-ce-mean-words.npz
[2019-12-07 13:54:54] [valid] Ep. 7 : Up. 130500 : ce-mean-words : 3.76078 : new best
[2019-12-07 13:54:56] Saving model weights and runtime parameters to model_gradual_2/ens/model.npz.best-perplexity.npz
[2019-12-07 13:55:00] [valid] Ep. 7 : Up. 130500 : perplexity : 42.9821 : new best
[2019-12-07 13:55:39] [valid] Ep. 7 : Up. 130500 : translation : 14.33 : stalled 2 times (last best: 14.46)

as the trainning process have more than 200k updates, Is it because the learning rate  became more and more smaller?
and how to set the learning rate schedule if the training have very big trainging dataset and updates maybe more than 300k to speed up the convergence?

Can I set the following parameter to alleviate the problem instead of epoch+stalled and increase the learing rate ?maybe 0.0006 or larger?
--lr-decay-strategy stalled 

thanks very much！

**Comment by wingsyuan** [comment]
Hi,
@snukky

In my new continue training script, I changed the parameters as follow:
--learn-rate 0.0003 --lr-warmup 150000 --lr-decay-inv-sqrt 150000 --lr-report \
(default parameters: --learn-rate 0.0003 --lr-warmup 16000 --lr-decay-inv-sqrt 16000 --lr-report \)

The purpose is to prevent the learning rate from dropping too quickly, and the training seem to be going well...are thert any side effects when I do this?

Thanks..

**Comment by snukky** [comment]
Smaller improvements on the validation set in further stages of training are expected. If your goal is to speed up the convergence, you may consider increasing the learning rate peak value, or postponing decaying, or both. Be careful as this may lead to a bad convergance or suboptimal performance.

I guess there is no simple answer to your questions. Optimal parameters for learning rate scheduling will vary among different languages, training data or even random model initializations. The default parameters in marian-examples should work fine in most cases.

**Comment by wingsyuan** [comment]
thanks a lot
@snukky
